tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,1310 @@
+                        setOutputType(optOutputType.get());
+                    else
+                    {
+                        const char* bundleAttr = stag.getValue("outputType");
+                        if (bundleAttr)
+                        {
+                            if (!setOutputType(bundleAttr))
+                                throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Unknown value '%s' provided for attribute Manifest/@outputType", bundleAttr);
+                        }
+                    }
+                    if (ManifestType::bundle == outputType)
+                        appendCompleteStartTag(MANIFEST_TAG_BUNDLE);
+
+                    do
+                    {
+                        type=xpp->next();
+                        switch (type)
+                        {
+                            case XmlPullParser::START_TAG:
+                            {
+                                xpp->readStartTag(stag);
+                                if (isManifestNodeNamed(stag, MANIFEST_TAG_SVC_BINDING))
+                                    type = processServiceBinding(xpp.get(), stag);
+                                else if (isManifestNodeNamed(stag, MANIFEST_TAG_SVC_DEFINITION))
+                                    type = processEsdlDefinition(xpp.get(), stag);
+                                else
+                                {
+                                    appendStartTag(xpp.get(), stag);
+                                    level++;
+                                }
+                                break;
+                            }
+
+                            case XmlPullParser::CONTENT:
+                                appendNonWhitespaceContent(xpp.get());
+                                break;
+
+                            case XmlPullParser::END_TAG:
+                                xpp->readEndTag(etag);
+                                if(!isRootEndTagNamed(etag, MANIFEST_TAG_MANIFEST, level))
+                                    appendEndTag(etag);
+                                level--;
+                                break;
+
+                            case XmlPullParser::END_DOCUMENT:
+                                level = 0;
+                                break;
+                        }
+
+                        lastType = type;
+
+                        #ifdef _MANIFEST_DEBUG_
+                        printf("processBinding while:(...%s)", output.length()<100 ? output.str() : output.str()+output.length()-100);
+                        #endif
+                    }
+                    while (level > 0);
+
+                    if (ManifestType::bundle == outputType)
+                        appendEndTag(MANIFEST_TAG_BUNDLE);
+                    if (ManifestType::bundle == outputType && !foundEsdlDefn)
+                        UWARNLOG("Generating EsdlBundle output without ESDL Definition. This may cause runtime issues.");
```

Comment:
Perhaps this should be an error?  I think all bundles should have interfaces defined?  If in the future we wanted to make it a warning we could.  Sometimes starting out strict is good.

Reply:
I'm fine with that. I'll plan to throw an exception to prevent the creation of a bundle without the interface- unless you meant just to use UERRLOG?

Reply:
Throw an exception, we can always relax it later if it is really needed.

Reply:
Fixed in today's commit.